src/Symfony/Component/Security/Core/Authentication/Token/SwitchUserToken.php

```diff
@@ -22,13 +24,13 @@ class SwitchUserToken extends UsernamePasswordToken
     private $originatedFromUri;
 
     /**
-     * @param string|object $user              The username (like a nickname, email address, etc.), or a UserInterface instance or an object implementing a __toString method
-     * @param mixed         $credentials       This usually is the password of the user
-     * @param string|null   $originatedFromUri The URI where was the user at the switch
+     * @param $user              The username (like a nickname, email address, etc.), or a UserInterface instance or an object implementing a __toString method
+     * @param $credentials       This usually is the password of the user
+     * @param $originatedFromUri The URI where was the user at the switch
```

Comment:
Is it valid to omit the type?

Reply:
for phpstorm at least it is

Reply:
https://docs.phpdoc.org/3.0/guide/references/phpdoc/tags/param.html

> ```
> @param [<Type>] [name] [<description>]
> ```
[…]
> When provided it MAY contain a Type to indicate what is expected.

Yes, the type is optional. But I had to look it up myself as well and I'm not yet sure if I like that notation. Maybe we just need to get used to it. 🤔

Reply:
If popular IDEs support it, I'm fine with it personally.

Reply:
Once you start using PHP attributes for everything ... you want to use it even to describe arguments. I wish this worked:

```php
#[Param('user', 'The username (like a nickname, email address, etc.), or a ...')]
#[Param('credentials', 'This usually is the password of the user')]
#[Param('originatedFromUri', 'The URI where was the user at the switch')]
public function __construct(string|\Stringable|UserInterface $user, mixed $credentials, /* ... */, string $originatedFromUri = null)
{
    // ...
}
```

Or even better, this:

```php
public function __construct(
    #[Param('The username (like a nickname, email address, etc.), or a ...')]
    private string|\Stringable|UserInterface $user,
    #[Param('This usually is the password of the user')]
    private mixed $credentials,
    // ...
    #[Param('The URI where was the user at the switch')]
    private string $originatedFromUri = null)
{
}
```

But support for attributes in PhpDocumentor doesn't seem a priority: https://github.com/phpDocumentor/Reflection/issues/185